lib/modules/platform/gitea/utils.ts

```diff
@@ -7,3 +15,50 @@ export function smartLinks(body: string): string {
 export function trimTrailingApiPath(url: string): string {
   return url?.replace(regEx(/api\/v1\/?$/g), '');
 }
+
+export function getRepoUrl(
+  repo: Repo,
+  gitUrl: GitUrlOption | undefined,
+  endpoint: string
+): string {
+  if (gitUrl === 'ssh') {
+    if (!repo.ssh_url) {
+      throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+    }
+    logger.debug({ url: repo.ssh_url }, `using SSH URL`);
+    return repo.ssh_url;
+  }
+
+  // Find options for current host and determine Git endpoint
+  const opts: HostRuleSearchResult = hostRules.find({
```

Comment:
```suggestion
  const opts = hostRules.find({
```
no explicit type required.